### Pytanie:
- Czy istnieje korelacja pomiędzy liczbą big maców, które może kupić obywatel, a przewidywaną długośćią życia?

#### Zaimportowanie potrzebnych bibliotek i wczytanie danych.

In [1]:
import pandas as pd
import numpy as np
big_mac = pd.read_csv("data/big-mac-adjusted-index.csv")
life_data = pd.read_csv("data/life_data.csv")

#### Przetwarzanie danych na temat Big Maca.

In [2]:
big_mac.columns

Index(['date', 'iso_a3', 'currency_code', 'name', 'local_price', 'dollar_ex',
       'dollar_price', 'GDP_dollar', 'adj_price', 'USD', 'EUR', 'GBP', 'JPY',
       'CNY'],
      dtype='object')

In [3]:
# usunięcie niepotrzebnych kolumn
big_mac.drop(['iso_a3','dollar_ex','adj_price','USD', 'EUR', 'GBP', 'JPY', 'CNY'], axis=1, inplace=True)

In [4]:
big_mac["date"] = pd.to_datetime(big_mac['date'], format='%Y-%m-%d')
big_mac.date.unique()

array(['2011-07-01T00:00:00.000000000', '2012-01-01T00:00:00.000000000',
       '2012-07-01T00:00:00.000000000', '2013-01-01T00:00:00.000000000',
       '2013-07-01T00:00:00.000000000', '2014-01-01T00:00:00.000000000',
       '2014-07-01T00:00:00.000000000', '2015-01-01T00:00:00.000000000',
       '2015-07-01T00:00:00.000000000', '2016-01-01T00:00:00.000000000',
       '2016-07-01T00:00:00.000000000', '2017-01-01T00:00:00.000000000',
       '2017-07-01T00:00:00.000000000', '2018-01-01T00:00:00.000000000',
       '2018-07-01T00:00:00.000000000', '2019-01-01T00:00:00.000000000',
       '2019-07-09T00:00:00.000000000', '2020-01-14T00:00:00.000000000'],
      dtype='datetime64[ns]')

In [5]:
big_mac = big_mac[(big_mac['date']>='2012-01-01') & (big_mac['date']<='2019-07-09')]

In [6]:
big_mac.head()

date currency_code       name  local_price  dollar_price  GDP_dollar
37 2012-01-01           ARS  Argentina        20.00      4.636606    9138.177
38 2012-01-01           AUD  Australia         4.80      4.936560   55589.549
39 2012-01-01           BRL     Brazil        10.25      5.678670   10816.487
40 2012-01-01           GBP    Britain         2.49      3.823395   36119.846
41 2012-01-01           CAD     Canada         4.73      4.632940   46214.914

In [7]:
# pogrupowanie danych według nazwy oraz roku 
big_mac = big_mac.groupby([big_mac['name'], big_mac['date'].dt.year]).mean()
big_mac.head(8)

local_price  dollar_price  GDP_dollar
name      date                                       
Argentina 2012         19.5      4.398785   9138.1770
          2013         20.0      3.848926  10959.0820
          2014         21.0      2.802453  11270.7810
          2015         28.0      3.158583  13790.8865
          2016         41.5      2.868772  13162.0210
          2017         62.5      3.796962  13045.8340
          2018         75.0      3.332768  13480.2305
          2019         97.5      2.436454  14525.4650

#### Przetwarzanie danych dotyczących przewidywanej długości życia.

In [8]:
life_data.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019'],
      dtype='object')

In [9]:
life_data.sample(3)

Country Name Country Code                           Indicator Name  \
172      Nigeria          NGA  Life expectancy at birth, total (years)   
171        Niger          NER  Life expectancy at birth, total (years)   
15       Belgium          BEL  Life expectancy at birth, total (years)   

     Indicator Code       1960       1961       1962       1963       1964  \
172  SP.DYN.LE00.IN  36.976000  37.431000  37.871000  38.291000  38.691000   
171  SP.DYN.LE00.IN  35.053000  35.141000  35.223000  35.298000  35.368000   
15   SP.DYN.LE00.IN  69.701951  70.520976  70.219512  70.051463  70.755122   

          1965  ...       2010       2011       2012       2013       2014  \
172  39.072000  ...  50.896000  51.346000  51.786000  52.228000  52.672000   
171  35.435000  ...  57.333000  58.081000  58.790000  59.455000  60.070000   
15   70.625366  ...  80.182927  80.585366  80.385366  80.587805  81.287805   

          2015       2016       2017  2018  2019  
172  53.112000  53.541000  53.950000   NaN   NaN  
171  60.631000  61.137000  61.599000   NaN   NaN  
15   80.992683  81.439024  81.439024   NaN   NaN  

[3 rows x 64 columns]

Brakuje danych na lata 2018-2019, w zwiąku z czym nie będą uwzględnione.

In [10]:
# usunięcie niepotrzebnych kolumn
life_data.drop(['Country Code', 'Indicator Name', 'Indicator Code','2018','2019'],axis=1, inplace=True)

In [11]:
life_data.columns

Index(['Country Name', '1960', '1961', '1962', '1963', '1964', '1965', '1966',
       '1967', '1968', '1969', '1970', '1971', '1972', '1973', '1974', '1975',
       '1976', '1977', '1978', '1979', '1980', '1981', '1982', '1983', '1984',
       '1985', '1986', '1987', '1988', '1989', '1990', '1991', '1992', '1993',
       '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002',
       '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011',
       '2012', '2013', '2014', '2015', '2016', '2017'],
      dtype='object')

In [12]:
# zamiana dat(z nazw kolumn na wiersze)
life_data = life_data.melt(id_vars='Country Name',var_name='date',value_name='life_expectancy')
life_data = life_data[life_data['date']>='2012']

In [13]:
life_data['date'] = life_data.date.astype(int)
life_data = life_data.rename(columns={'Country Name': 'name'})
# pogrupowanie danych ze względu na nazwę oraz datę
life_data = life_data.groupby(['name','date']).mean()

#### Połączenie danych na temat Big Maca i długości życia.

In [14]:
big_mac = big_mac.merge(life_data,on=['name','date'])

In [15]:
big_mac = big_mac.reset_index()

In [16]:
# obliczenie liczby big maców jakie może kupić jedna osoba
big_mac['number_of_big_mac'] = big_mac.GDP_dollar/big_mac.dollar_price

In [17]:
big_mac.sample(5)

name  date  local_price  dollar_price  GDP_dollar  \
102  New Zealand  2012          5.1      4.026068  32145.2330   
163  Switzerland  2013          6.5      6.921945  83072.8450   
108       Norway  2012         42.0      6.923286  84443.6340   
65       Hungary  2017        881.0      3.131879  12509.0910   
89         Japan  2017        380.0      3.309118  35701.4155   

     life_expectancy  number_of_big_mac  
102        81.156098        7984.275722  
163        82.797561       12001.373160  
108        81.451220       12197.046136  
65         76.063415        3994.116317  
89         84.099756       10788.799963

In [18]:
# obliczenie korelacji pomiędzy poszczególnymi cechami
big_mac.corr()

date  local_price  dollar_price  GDP_dollar  \
date               1.000000     0.019044     -0.091303    0.004629   
local_price        0.019044     1.000000     -0.193437   -0.219537   
dollar_price      -0.091303    -0.193437      1.000000    0.803161   
GDP_dollar         0.004629    -0.219537      0.803161    1.000000   
life_expectancy    0.067050    -0.220608      0.632126    0.730135   
number_of_big_mac  0.048432    -0.250318      0.595074    0.929259   

                   life_expectancy  number_of_big_mac  
date                      0.067050           0.048432  
local_price              -0.220608          -0.250318  
dollar_price              0.632126           0.595074  
GDP_dollar                0.730135           0.929259  
life_expectancy           1.000000           0.766016  
number_of_big_mac         0.766016           1.000000

Jak widać w powyższej tabeli korelacja pomiędzy liczbą Big Maców, które może kupić obywatel, a przewidywaną długością życia jest wysoka (0.766016),  jednak warto zwrócić uwage również na wysoką korelację pomiędzy przewidywaną długością życia, a PKB per capita (0.730135) oraz bardzo silną korelację pomiędzy PKB, a liczbą Big Maców, które może kupić obywatel (0.929259). Jak widać korelacja pomiędzy PKB a Big Macami jest niemal pewna.

Wniosek:
Długość życia jest silnie zależna od zamożności państwa. Big Mac index to wskaźnik pomiaru parytetu siły nabywczej. Posiadając podstwową wiedzę na temat zdrowego żywienia można stwierdzić, że spożywanie dużej ilości Big Maców zdecydowanie nie wpłya korzystnie na stan zdrowia oraz długość życia.

##### Źródła:

Big Mac index:
https://github.com/TheEconomist/big-mac-data/tree/master/output-data

Przewidywana długość życia w chwili urodzenia:
http://api.worldbank.org/v2/en/indicator/SP.DYN.LE00.IN?downloadformat=csv


Autor:
Marcin Mulawa